sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/functions/FunctionsTest.java

```diff
@@ -349,6 +386,19 @@ public void testDecimalErrors() {
                 // The compiler rounds `1234.1236` to `1234.124` before calling the rust code
                 "cannot represent 1234.124 as DECIMAL(6, 3)"
         );
+
+        this.shouldFail("select cast(1234.1234 AS DECIMAL(6, 3))",
```

Comment:
maybe the error message should be more elaborate: precision of result is to small to accommodate value